### DISCUSSION TITLE
Where is the 'include' directory

### DISCUSSION THREAD

**Discussion by wojtase1** [PySlurm/pyslurm#275] [Q&A]
I am trying to get slurm running on a cluster of raspberry pis.  I have installed slurm and got it running and wanted to experiment with this python library.  So I downloaded the library and when I tried to install it I found that the includes and libslurm.so were not installed?  Seemed strange so i downloaded the slurm source code compiled it and was able to create libslurm.so but now when trying to install pyslurm it wants the include directory?  What include directory is it looking for.  I got a warning that it could not find 
"slurm_version.h" and I have searched my entire machine and I am unable to find it.  Any ideas?

**Comment by tazend** [comment]
Hi @wojtase1 

By default, we scan for libslurm.so in `/usr/lib64` and in `/usr/include/` for the headers. Both of these are necessary to compile pyslurm. However, if Slurm is installed in a different location, then it will be necessary to instruct pyslurm where it can find libslurm.so and the slurm headers (like slurm_version.h, slurm.h, slurmdbd.h, ...)

For example, if you compiled and installed Slurm from source like this:

```bash
cd slurm-src
./configure --prefix=/opt/slurm
make install
```

Then you will need to tell pyslurm that the slurm installation can be found in `/opt/slurm`, for example by defining these two environment variables prior to compiling pyslurm:

```bash
export SLURM_INCLUDE_DIR=/opt/slurm/include
export SLURM_LIB_DIR=/opt/slurm/lib
```

Hope that helps :)

**Reply by wojtase1** [reply]
Bah!  I had a newb problem.  I did not match up the versions of the library to the version of slurm that I was building to which figures why it was not able to find certain files.

Thanks.
